src/app/components/pages/quizzes/MyQuizzes.tsx

```diff
@@ -222,82 +313,117 @@ const MyQuizzesPageComponent = ({user}: MyQuizzesPageProps) => {
+    //         falseLabel="Table"
+    //         trueLabel="Cards"
+    //         onChange={() => setDisplayMode(d => d === "table" ? "cards" : "table")}
+    //     />
+    const displayModeToggle = <div className="d-flex flex-column align-items-start pb-3 pe-3 col-8 col-sm-6 col-md-3">
+        <span>Display in</span>
+        <StyledDropdown value={displayMode} onChange={() => setDisplayMode(d => d === "table" ? "cards" : "table")}>
+            <option value="table">Table View</option>
+            <option value="cards">Card View</option>
+        </StyledDropdown>
+    </div>;
+
+    const pastTestsToggle = <div className="d-flex flex-column align-items-center w-max-content pb-3">
+        <span>Past tests</span>
+        <div className="h-100 align-content-center">
+            <StyledToggle
+                checked={showCompleted}
+                falseLabel="Hidden"
+                trueLabel="Shown"
+                onChange={() => {
+                    const target = !showCompleted;
+                    setShowCompleted(target);
+                    setQuizStatuses(s => target ? [...s, QuizStatus.Complete, QuizStatus.Overdue] : s.filter(status => ![QuizStatus.Complete, QuizStatus.Overdue].includes(status)));
+                }}
+            />
+        </div>
+    </div>;
+
+    // +!! converts a string to 0 if null or empty and 1 otherwise
+    const filterCount = +!!quizTitleFilter + +!!quizCreatorFilter + quizStatusFilter.length;
+
+    const filtersToggle = <Col xs={3} sm={2} md={1} className="pb-3 ms-3">
+        <Label className="w-100 d-flex flex-column align-items-center mb-0">
+            <span className="text-nowrap">
+                Filters
+                {<FilterCount count={filterCount ?? 0} className="ms-2 mb-1" />}
+            </span>
+            <Button color="secondary" className={classNames("w-100 gameboards-filter-dropdown align-self-center", {"selected": showFilters})}
+                onClick={() => setShowFilters(s => !s)} data-testid="filter-dropdown"/>
+        </Label>
+    </Col>;
+
+    const tabTopContent = <>
+        <div className="d-flex">
+            {displayModeToggle}
+            <Spacer/>
+            {above["sm"](deviceSize) && pastTestsToggle}
+            {filtersToggle}
+        </div>
+        <QuizFilters setShowCompleted={setShowCompleted} setQuizCreator={setQuizCreator} setQuizTitleFilter={setQuizTitleFilter} 
+            quizStatuses={quizStatusFilter} setQuizStatuses={setQuizStatuses} showFilters={showFilters}/>
+    </>;
+
+    const emptyAssignedMessage = <span className="text-muted">{!quizAssignments || quizAssignments.length === 0
+        ? "You have no tests in progress."
+        : "No tests match your filters."
+    }</span>;
 
-    return <RS.Container>
+    const emptyPracticeMessage = <span className="text-muted">
+        No practice tests match your filters. Take some new tests <Link to="/practice_tests">here</Link>!
+    </span>;
```

Comment:
This link appears on both sites, but `/practice_tests` doesn't exist on Ada.

I also like the distinction between "no tests at all" and "no filtered tests" as in `emptyAssignedMessage`, but a nested conditional here probably isn't the best solution.

```suggestion
        const emptyPracticeMessage = <span className="text-muted">{isPhy
        ? !freeAttempts || freeAttempts.length === 0
            ? <>You have no practice tests in progress. Take some new tests <Link to="/practice_tests">here</Link>!</>
            : <>No practice tests match your filters. Take some new tests <Link to="/practice_tests">here</Link>!</>
        : !freeAttempts || freeAttempts.length === 0
            ? <>You have no practice tests in progress.</>
            : <>No practice tests match your filters.</>
    }</span>;
```

Reply:
Ah, it is supposed to exist on Ada. That simplifies the conditional enough that it feels nice. I'm also inclined to only have the link on if there are no tests at all, as if you're using the filters you're there to find an existing test, not take a new one.